internal/provider/resource_device.go

```diff
@@ -0,0 +1,215 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/paultyng/go-unifi/unifi"
+)
+
+func resourceDevice() *schema.Resource {
+	return &schema.Resource{
+		Description: "`unifi_device` manages a device of the network.\n\n" +
+			"Devices are adopted by the controller, so it may not be possible " +
+			"for this resource to be created through terraform.",
+
+		//Create: resourceDeviceCreate,
+		Read:   resourceDeviceRead,
+		Update: resourceDeviceUpdate,
+		Delete: resourceDeviceDelete,
+		Importer: &schema.ResourceImporter{
+			State: importSiteAndID,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The ID of the device.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"site": {
+				Description: "The name of the site to associate the device with.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"mac": {
+				Description:      "The MAC address of the device.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: macDiffSuppressFunc,
+				ValidateFunc:     validation.StringMatch(macAddressRegexp, "Mac address is invalid"),
+			},
+			"name": {
+				Description: "The name of the device.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"disabled": {
+				Description: "Specifies whether this device should be disabled.",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"port_overrides": {
+				Description: "Settings overrides for specific switch ports.",
+				Type:        schema.TypeList,
```

Comment:
I think you probably want this to be a `TypeSet` since the order won't matter.

Reply:
Changed to TypeSet